wb/nm_helper/connection_manager.py

```diff
@@ -257,14 +261,76 @@ def read_config_json():
         return json.load(file)
 
 
+def wait_pycares_channel(channel):
+    start = time.monotonic_ns()
+    while time.monotonic_ns() - start < 5000000000:
+        read_fds, write_fds = channel.getsock()
+        if not read_fds and not write_fds:
+            break
+        timeout = channel.timeout()
+        no_file_d = pycares.ARES_SOCKET_BAD  # pylint: disable=E1101
+        if not timeout:
+            channel.process_fd(no_file_d, no_file_d)
+            continue
+        rlist, wlist, _xlist = select.select(read_fds, write_fds, [], timeout)
+        for file_d in rlist:
+            channel.process_fd(file_d, no_file_d)
+        for file_d in wlist:
+            channel.process_fd(no_file_d, file_d)
+
+
+class PycaresCallback:  # pylint: disable=R0903
+    def __init__(self) -> None:
+        self.result = None
+        self.error = None
+
+    def __call__(self, result, error) -> None:
+        self.result = result
+        self.error = error
+
+
+def resolve_domain_name(name: str, iface: str) -> str:
+    channel = pycares.Channel()
+    channel.set_local_dev(iface.encode())
+    callback = PycaresCallback()
+    channel.gethostbyname(name, socket.AF_INET, callback)
```

Comment:
Мне кажется, инлайн функция тут была бы более наглядной (а ещё более наглядным тут было бы asyncio, ну да ладно)

```suggestion
    callback_result = None
    callback_error = None
    def callback(result, error):
      callback_result, callback_error = result, error
      
    channel.gethostbyname(name, socket.AF_INET, callback)
```

Reply:
Я пробовал лямбду, но оказалось, что они в питоне не умеют захватывать контекст. Про то, что локальные функции видят переменные родителя, я не знал. Переделаю